doc/tutorials/content/gpu_octree.rst

```diff
@@ -0,0 +1,235 @@
+
+In contrast to the their CPU counterparts, all of these functions accept a vector of queries rather than a single query. Apart from this, there are some minor differences between the CPU and GPU implementations, and these will be highlighted throughout this tutorial.
+
+.. note::
+
+    The above functions may give erroneous results in PCL versions < 1.12 due to warp synchronization errors. These errors have been fixed from PCL 1.12 upwards.
+
+.. note::
+
+    The API of the above three functions have been modified from PCL 1.12 onwards to return squared distances from query point to result points. The following tutorial is based on the modified API. You can reference the API of the GPU octree search methods `here <https://pointclouds.org/documentation/classpcl_1_1gpu_1_1_octree.html>`_.
+
+The following table shows a sample comparison between the performance of the CPU and GPU implementations.
+
+
+The code
+========
+First, create a file, let’s say, gpu_octree_search.cpp and place the following code inside it:
+
+.. literalinclude:: sources/gpu_octree/gpu_octree.cpp
+    :language: cpp
+    :linenos:
+
+The Explanation
+===============
+
+Initially, we define and instantiate a PointCloud structure and fill it with random points. We also create some random queries.
+
+.. literalinclude:: sources/gpu_octree/gpu_octree.cpp
+    :language: cpp
+    :lines: 22-45
+
+
+Building an octree
+------------------
+
+Next we create an octree instance and build the octree. In order to do this, we must first create an instance of `pcl::gpu::Octree::PointCloud` using the points in our point cloud. In order to do this, we must first upload those points to the GPU. Here "upload" refers to copying data from host memory to GPU memory (Conversely, "download" refers to copying data from GPU memory to host memory). We will also upload our queries for later use.  For this, we must create a `pcl::gpu::Octree::Queries` object and upload query points to this object.
+
+Note that unlike in the CPU octree implementation, the user cannot set the resolution of the octree. Instead, the depth of the octree will be automatically determined depending on the point distribution of the cloud (However, the maximum depth is capped at 10).
+
+.. literalinclude:: sources/gpu_octree/gpu_octree.cpp
+    :language: cpp
+    :lines: 47-58
+
+After building the octree, we we can move on to the different search queries.
+
+K Nearest Neighbors search
+---------------------------
+
+The K nearest neighbor algorithm is designed to identify a given (k) number of nearest neighbors to a query point. However, the GPU KNN search algorithm is currently restricted to k=1 only.
+
+
+.. note::
+
+    The GPU K nearest neighbor algorithm differs from it's CPU counterpart in implementation. Specifically, the CPU algorithm will only find approximate nearest neighbors, where the user can set the upper bound of the error allowed. In contrast, the GPU algorithm finds the absolute nearest neighbor.
```

Comment:
Probably also add that this the only reason we're currently offering the kNearestNeighbor despite it only supporting k=1. Otherwise I would normally route people to using approxNearestNeighbor.